### PR TITLE
Add attributes for Sp and MVBS from index binning

### PR DESCRIPTION
**Pull request by emiliom**
Addresses remaining tasks in #582 and #481

### CODE REVIEW

File: echopype/preprocess/api.py [hunk 1]
```diff
@@ -156,29 +170,32 @@ def compute_MVBS_index_binning(ds_Sv, range_sample_num=100, ping_num=100):
...
-    # Attach attributes
-    ds_out.attrs = {
-        "binning_mode": "index",
-        "range_sample_num": range_sample_num,
-        "ping_num": ping_num,
-    }
+    _set_MVVBS_attrs(ds_MVBS)
+    ds_MVBS["Sv"] = ds_MVBS["Sv"].assign_attrs(
+        {
+            "comment": "MVBS binned on the basis of range_sample and ping number specified as index numbers",
```
Comment (emiliom) [inline echopype/preprocess/api.py:191]: @leewujung pay attention to this addition. I couldn't come up with a `cell_methods` alternative. I was getting hung up with the specification of bins based on index numbers rather than physical units.
FYI, `comment` is a CF attribute.

Comment (emiliom) [inline echopype/preprocess/api.py:191]: @leewujung pay attention to this addition. I couldn't come up with a `cell_methods` alternative. I was getting hung up with the specification of bins based on index numbers rather than physical units.
FYI, `comment` is a CF attribute.

File: echopype/preprocess/api.py [hunk 2]
```diff
@@ -16,6 +16,30 @@ def _check_range_uniqueness(ds):
     ).all()
 
 
+def _set_MVVBS_attrs(ds):
```
Comment (leewujung) [inline echopype/preprocess/api.py:19]: noticed a small typo "MVVBS" here

Comment (emiliom) [reply echopype/preprocess/api.py:19]: Good eye. I'll fix it

File: echopype/preprocess/api.py (hunk 1 above)
Comment (leewujung) [reply echopype/preprocess/api.py:191]: Based on what you had for the case when MVBS is computed based on physical units (actual range in meters and ping time), how about something like below?
```
{
            "cell_methods": (
                f"ping_time: mean (interval: {ping_num} pings "
                "comment: ping_time is the interval start) "
                f"echo_range: mean (interval: {range_sample_num} samples along range "
                "comment: echo_range is the interval start)"
            ),
            "binning_mode": "sample number",
            "range_meter_interval": str(range_sample_num) + "samples along range",
            "ping_time_interval": str(ping_num) + "pings",
}
```

File: echopype/preprocess/api.py (hunk 2 above)
Comment (emiliom) [reply echopype/preprocess/api.py]: It's fixed.

File: echopype/preprocess/api.py (hunk 1 above)
Comment (emiliom) [reply echopype/preprocess/api.py:191]: Adding link to where I discussed CF `cell_methods`: https://github.com/OSOceanAcoustics/echopype/pull/594#pullrequestreview-921175073

Comment (emiliom) [reply echopype/preprocess/api.py:191]: A challenge with your suggestion is the interval units string. [Per CF](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.9/cf-conventions.html#recording-spacing-original-data): 
> The syntax is (`interval`: *value unit*), where *value* is a numerical value and *unit* is a string that can be recognized by UNIDATA’s Udunits package [[UDUNITS]](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.9/cf-conventions.html#UDUNITS).

So, strictly speaking, using "pings" and "samples along range" would not be compliant. The difficulty is in trying to specify a constant interval that's based on the index number rather than the dimension units. Though CF also says this:

> The *unit* will usually be dimensionally equivalent to the unit of the corresponding dimension, but this is not required (which allows, for example, the interval for a standard deviation calculated from points evenly spaced in distance along a parallel to be reported in units of length even if the zonal coordinate of the cells is given in degrees).

Let me chew on this.

Comment (emiliom) [reply echopype/preprocess/api.py:191]:
```
"range_meter_interval": str(range_sample_num) + "samples along range",
"ping_time_interval": str(ping_num) + "pings",
```
This seems confusing, since the value after "range_meter_interval" will not be in meters, nor will the value after "ping_time_interval" be in time units. It seems like the goal of these custom echopype attributes was to record the function parameters used, as is. If we find a way to settle on using `cell_methods`, the linkage between `ping_time` and `echo_range` and the binning method (mean) & interval values used would be explicit, so I'd then vote for retaining the original attribute names of "range_sample_num" and "ping_num".

Comment (leewujung) [reply echopype/preprocess/api.py:191]: I am not entirely understanding what you are proposing, but when proposing the `cell_methods` I actually forgot to check the attribute names. It should be
```
"range_sample_interval": str(range_sample_num) + "samples along range",
"ping_interval": str(ping_num) + "pings",
```
ie "range_**sample**_interval" instead of "range_meter_interval", "ping_~time~_interval" instead of "ping_time_interval"

**Comment by emiliom** [comment]
Pushed new commits:
- Replace Sp with TS. See https://github.com/OSOceanAcoustics/echopype/issues/582#issuecomment-1095258029
- Update dB reference levels and symbology. See https://github.com/OSOceanAcoustics/echopype/issues/582#issuecomment-1094504660
- Add MVBS_index_binning cell_methods attribute and update other provenance attributes

**Comment by leewujung** [comment]
Great, thanks @emiliom !
The failed pre-commit check are just 2 lines that are too long:
https://github.com/OSOceanAcoustics/echopype/blob/dd369c1282abfdbea6cc479f6247e591cab69678/echopype/preprocess/api.py#L131
and
https://github.com/OSOceanAcoustics/echopype/blob/dd369c1282abfdbea6cc479f6247e591cab69678/echopype/preprocess/api.py#L197
For readability an `#noqa` may be in order.

File: echopype/calibrate/api.py [hunk 3]
```diff
@@ -49,29 +49,36 @@ def _compute_cal(
...
-        sv_dataset["range_sample"].attrs = {"long_name": "Along-range bin (sample) number, base 0"}
-        sv_dataset["echo_range"].attrs = {"long_name": "Range distance", "units": "m"}
 
+    # Perform calibration
+    if cal_type == "Sv":
+        sv_dataset = cal_obj.compute_Sv(waveform_mode=waveform_mode, encode_mode=encode_mode)
+        add_attrs("Sv", sv_dataset)
         if "water_level" in echodata.platform.data_vars.keys():
             # add water_level to the created xr.Dataset
             sv_dataset["water_level"] = echodata.platform.water_level
```
Comment (leewujung) [inline echopype/calibrate/api.py:77]: @emiliom : Not sure if this should be in another PR: 

Looking at this now, I realized the check and addition of `water_level` should be in the outputs for both compute_Sv and compute_TS so that users can use this to convert `echo_range` to `depth` easily. 

Also in the realm of this PR, the attributes of `water_level` can be set in `_set_MVBS_attrs` below under echopype/preprocess/api.py.

Comment (leewujung) [reply echopype/calibrate/api.py:77]: Related to #516 #583 #586

Comment (emiliom) [reply echopype/calibrate/api.py:77]: It could be a separate PR. But this looks like an easy change w/o additional ramifications that will lead to protracted discussions. It's also in line with the broader improvements to `compute_MVBS_index_binning`. So, I'll just add it to this PR.

Comment (leewujung) [reply echopype/calibrate/api.py:77]: Sounds good. Thank you @emiliom !

Comment (emiliom) [reply echopype/calibrate/api.py:77]: > Also in the realm of this PR, the attributes of water_level can be set in _set_MVBS_attrs below under echopype/preprocess/api.py.

Actually, `water_level` retains its original attributes from the echodata `Platform` group. So, no attribute assignment is taking place in `calibrate/api.py`.

I've implemented the addition of `water_level` to `compute_TS` (and in the process it made the code quite a bit simpler!). I'll push it to the PR in a bit.

**Comment by emiliom** [comment]
> For readability an #noqa may be in order.

Done

**Comment by leewujung** [comment]
I think you had a typo on noqa 🤪

**Comment by emiliom** [comment]
> I think you had a typo on noqa

:weary: Sigh. I'll fix it now.

**Comment by emiliom** [comment]
pre-commit is now passing. Nice!

**Comment by leewujung** [comment]
Thanks @emiliom ! I'll add a small thing about TS applicability in the docstring later today, and then merge this.

**Comment by emiliom** [comment]
I think what remains an open question is the `cell_methods` implementation. As I wrote earlier (above):

> A challenge with your suggestion is the interval units string. [Per CF](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.9/cf-conventions.html#recording-spacing-original-data): 
>> The syntax is (`interval`: *value unit*), where *value* is a numerical value and *unit* is a string that can be recognized by UNIDATA’s Udunits package [[UDUNITS]](http://cfconventions.org/Data/cf-conventions/cf-conventions-1.9/cf-conventions.html#UDUNITS).
>
> So, using "pings" and "samples along range" would not be CF compliant. The difficulty is in trying to specify a constant interval that's based on the index number rather than the dimension units. 

I don't have a solution. I think we should just merge this PR (after your additions), then come back to it in the future.

**Comment by leewujung** [comment]
@emiliom : I think you should make a decision here since I am not familiar with the CF conventions here. 

I will merge this after my addition of TS-related docstring.

**Comment by leewujung** [comment]
@emiliom : do you wanna capture this leftover dilemma in a new issue? Thanks!

**Comment by emiliom** [comment]
> @emiliom : do you wanna capture this leftover dilemma in a new issue? Thanks

Done. #629
